### DISCUSSION TITLE
Configuration ignore rssi, change model type, whitelist/blacklist

### DISCUSSION THREAD

**Discussion by johannes-z** [theengs/gateway#238] [Q&A]
Hello! I ran into a few problems running TheengsGateway and hope somebody can help me out.

I have 3 devices connected to a raspberry pi 4:
* 2x Xiaomi MiFlora (`HHCCJCY01HHCC`)
* 1x SwitchBot Thermometer and Hygrometer (`THX1(W230150X)`)

1. When running TheengsGateway in active mode, my mqtt is *spammed* with dozens of message because the rssi keeps updating multiple times per second. I fear this may drain the battery of the bluetooth devices. Furthermore, `ble_scan_time` and `ble_time_between_scans` is both set to 5, but it doesn't seem to be honored as it keeps updating as fast as possible.

    a. __I tried passive mode which works fine, but the SwitchBot device isn't picked up anymore. It seems this device requires active scanning?__ I get this data:

```
DEBUG:BLEGateway:11:22:33:44:55:66:AdvertisementData(local_name='WoSensorTHP', manufacturer_data={2409: b'\xf3\xa5]\x0eW!_\x01\x04\x96)'}, service_uuids=['00001800-0000-1000-8000-00805f9b34fb', '00001801-0000-1000-8000-00805f9b34fb', 'cba20d00-224d-11e6-9fb8-0002a5d5c51b'], rssi=-70)
DEBUG:BLEGateway:None
DEBUG:BLEGateway:Sent `{"manufacturerdata": "6909f3a55d0e57215f01049629", "name": "WoSensorTHP", "id": "11:22:33:44:55:66", "rssi": -70, "mfr": "Woan Technology (Shenzhen) Co., Ltd."}` to topic `home/TheengsGateway/BTtoMQTT/112233445566`
```

2. The Xiaomi MiFlora is picked up as model `HHCCJCY01HHCC`, but it should be `HHCCJCY10`. I had it connected to home assistant directly before and I got the battery state, which is now missing. Is there anything I can do here? 

3. I would like to limit TheengsGateway to only the 3 MAC addresses of the devices listed above, and not publish anything else, such as phones or TV devices.

**Comment by DigiH** [comment]
> 2. The Xiaomi MiFlora is picked up as model `HHCCJCY01HHCC`, but it should be `HHCCJCY10`. I had it connected to home assistant directly before and I got the battery state, which is now missing. Is there anything I can do here?

Could you post some MQTT message data with 

> -padv PUBLISH_ADVDATA, --publish_advdata PUBLISH_ADVDATA
>                         Publish advertising and advanced data (1) or not (0)
>                         (default: 0)

set to 1 for your HHCCJCY10?

**Reply by johannes-z** [reply]
```
DEBUG:BLEGateway:XX:XX:XX:XX:XX:XX:AdvertisementData(local_name='Flower care', service_data={'0000fe95-0000-1000-8000-00805f9b34fb': b'q \x98\x00\xf0\x0b\x0b\xb1~\x85\\\r\x08\x10\x01"'}, service_uuids=['0000fe95-0000-1000-8000-00805f9b34fb'], rssi=-84)
DEBUG:BLEGateway:{"name":"Flower care","id":"XX:XX:XX:XX:XX:XX","rssi":-84,"servicedatauuid":"fe95","servicedata":"71209800f00b0bb17e855c0d08100122","brand":"Xiaomi/VegTrug","model":"MiFlora","model_id":"HHCCJCY01HHCC","type":"PLANT","moi":34,"mac":"XX:XX:XX:XX:XX:XX"}
INFO:BLEGateway:publishing device `{'name': 'Flower care', 'id': 'XX:XX:XX:XX:XX:XX', 'rssi': -84, 'brand': 'Xiaomi/VegTrug', 'model': 'MiFlora', 'model_id': 'HHCCJCY01HHCC', 'type': 'PLANT', 'moi': 34, 'mac': 'XX:XX:XX:XX:XX:XX'}`
DEBUG:BLEGateway:Sent `{"stat_t": "+/+/BTtoMQTT/XXXXXXXXXXXX", "dev_cla": "illuminance", "unit_of_meas": "lx", "state_class": "measurement", "name": "HHCCJCY01HHCC-lux", "uniq_id": "XXXXXXXXXXXX-lux", "val_tpl": "{{ value_json.lux | is_defined }}", "device": {"ids": ["XXXXXXXXXXXX"], "cns": [["mac", "XXXXXXXXXXXX"]], "mf": "Xiaomi/VegTrug", "mdl": "HHCCJCY01HHCC", "name": "MiFlora-B10B0B", "via_device": "TheengsGateway"}}` to topic `homeassistant/sensor/XXXXXXXXXXXX-lux/config`
DEBUG:BLEGateway:Sent `{"stat_t": "+/+/BTtoMQTT/XXXXXXXXXXXX", "dev_cla": "temperature", "unit_of_meas": "\u00b0C", "state_class": "measurement", "name": "HHCCJCY01HHCC-tempc", "uniq_id": "XXXXXXXXXXXX-tempc", "val_tpl": "{{ value_json.tempc | is_defined }}", "device": {"ids": ["XXXXXXXXXXXX"], "cns": [["mac", "XXXXXXXXXXXX"]], "mf": "Xiaomi/VegTrug", "mdl": "HHCCJCY01HHCC", "name": "MiFlora-B10B0B", "via_device": "TheengsGateway"}}` to topic `homeassistant/sensor/XXXXXXXXXXXX-tempc/config`
DEBUG:BLEGateway:Sent `{"stat_t": "+/+/BTtoMQTT/XXXXXXXXXXXX", "unit_of_meas": "\u00b5S/cm", "state_class": "measurement", "name": "HHCCJCY01HHCC-fer", "uniq_id": "XXXXXXXXXXXX-fer", "val_tpl": "{{ value_json.fer | is_defined }}", "device": {"ids": ["XXXXXXXXXXXX"], "cns": [["mac", "XXXXXXXXXXXX"]], "mf": "Xiaomi/VegTrug", "mdl": "HHCCJCY01HHCC", "name": "MiFlora-B10B0B", "via_device": "TheengsGateway"}}` to topic `homeassistant/sensor/XXXXXXXXXXXX-fer/config`
DEBUG:BLEGateway:Sent `{"stat_t": "+/+/BTtoMQTT/XXXXXXXXXXXX", "unit_of_meas": "%", "state_class": "measurement", "name": "HHCCJCY01HHCC-moi", "uniq_id": "XXXXXXXXXXXX-moi", "val_tpl": "{{ value_json.moi | is_defined }}", "device": {"ids": ["XXXXXXXXXXXX"], "cns": [["mac", "XXXXXXXXXXXX"]], "mf": "Xiaomi/VegTrug", "mdl": "HHCCJCY01HHCC", "name": "MiFlora-B10B0B", "via_device": "TheengsGateway"}}` to topic `homeassistant/sensor/XXXXXXXXXXXX-moi/config`
DEBUG:BLEGateway:Sent `{"name": "Flower care", "id": "XX:XX:XX:XX:XX:XX", "rssi": -84, "brand": "Xiaomi/VegTrug", "model": "MiFlora", "model_id": "HHCCJCY01HHCC", "type": "PLANT", "moi": 34, "mac": "XX:XX:XX:XX:XX:XX"}` to topic `home/TheengsGateway/BTtoMQTT/XXXXXXXXXXXX`
```

I started it with `TheengsGateway --publish_advdata 1`, but the result seems to be the same.

**Reply by DigiH** [reply]
> I started it with `TheengsGateway --publish_advdata 1`, but the result seems to be the same.

Hmmm, something is not right then, could you try starting with 

`python -m TheengsGateway -padv 1`

The last Flower care message should then have its servicedata included.

**Actually** the first Flower Care message has it ;) 

DEBUG:BLEGateway:{"name":"Flower care","id":"XX:XX:XX:XX:XX:XX","rssi":-84,"servicedatauuid":"fe95","servicedata":"71209800f00b0bb17e855c0d08100122","brand":"Xiaomi/VegTrug","model":"MiFlora","model_id":"HHCCJCY01HHCC","type":"PLANT","moi":34,"mac":"XX:XX:XX:XX:XX:XX"}

but this is showing the `servicedatauuid` usually associated with HHCCJCY01HHCC and also the  `servicedata` being much longer that the usual HHCCJCY10 servicedata, possibly also correct for the various HHCCJCY10 broadcasts.

Could you do the same as I suggested with the SwitchBot broadcasts, monitor the messages over some time in the History of MQTT Explorer? To see if all the HHCCJCY10 braodcasts are with "servicedatauuid":"fe95" and longer actual servicedata?

**Reply by johannes-z** [reply]
I ran it again with `-padv 1`.

I'm not sure how long the `servicedata` should normally be there, but it's present on all messages so far, here are some examples of the values:

```
  "servicedatauuid": "fe95",
  "servicedata": "712098000d0b0bb17e855c0d091002c301",

  "servicedatauuid": "fe95",
  "servicedata": "712098000f0b0bb17e855c0d071003560000",

  "servicedatauuid": "fe95",
  "servicedata": "71209800110b0bb17e855c0d091002c301",

  "servicedatauuid": "fe95",
  "servicedata": "71209800130b0bb17e855c0d071003560000",

  "servicedatauuid": "fe95",
  "servicedata": "71209800150b0bb17e855c0d091002c101",

  "servicedatauuid": "fe95",
  "servicedata": "71209800170b0bb17e855c0d071003560000",

  "servicedatauuid": "fe95",
  "servicedata": "71209800190b0bb17e855c0d091002c301",

  "servicedatauuid": "fe95",
  "servicedata": "712098001b0b0bb17e855c0d0710036d0000",
```

**Reply by DigiH** [reply]
Is it the pink, supposedly Tuya, version of the HHCC Mi Flora you have?

As when previously creating a decoder for it the serviceuuid and datalength of the servicedata was quite different to what you're posting above.

https://community.openmqttgateway.com/t/feature-request-hhcc-flora-care-senosr-pink-version-tuya/2313

You samples do very much look like HHCCJCY01HHCC, or it could well be that it is even a different version of the Mi Flora, for which we haven;'t got a decoder yet, so that it might fall into the wrong one.

Or do you actually have HHCCJCY01HHCCs?

> I have 3 devices connected to a raspberry pi 4:
> 
>     2x Xiaomi MiFlora (HHCCJCY01HHCC)

I suppose you are not getting any messages with  "servicedatauuid": "fd50" and shorter servicedata.

**Reply by johannes-z** [reply]
I have the typical white sensors. They seem to be offbrand HHCC devices, more specifically `B BestCon HHCC`; I got them off amazon. I will check the servicedata tomorrow.
